### ISSUE TITLE
Ambiguous match found exception on BulkMerge

### ISSUE THREAD

**Issue by Borja204** [zzzprojects/EntityFramework-Extensions#106]
Hello,

I'm trying to use BulkMerge. It works well with some of the tables but it fails when I try to merge with the Users table (the Identity's table).

The stacktrace is the following:

` en System.RuntimeType.GetPropertyImpl(String name, BindingFlags bindingAttr, Binder binder, Type returnType, Type[] types, ParameterModifier[] modifiers)
   en System.Type.GetProperty(String name, BindingFlags bindingAttr)
   en Z.EntityFramework.Extensions.Core.Mapping.ScalarAccessorMapping.(Object )
   en Z.EntityFramework.Extensions.Core.Mapping.ScalarAccessorMapping.(Object )
   en .[](DbContext this, List`1 ,  , SchemaEntityType , Int32 , List`1 , Boolean )
   en DbContextExtensions.`1.( )
   en .Read()
   en .()
   en .Read()
   en System.Data.SqlClient.SqlBulkCopy.ReadFromRowSource()
   en System.Data.SqlClient.SqlBulkCopy.ReadFromRowSourceAsync(CancellationToken cts)
   en System.Data.SqlClient.SqlBulkCopy.WriteToServerInternalAsync(CancellationToken ctoken)
   en System.Data.SqlClient.SqlBulkCopy.WriteRowSourceToServerAsync(Int32 columnCount, CancellationToken ctoken)
   en System.Data.SqlClient.SqlBulkCopy.WriteToServer(IDataReader reader)
   en .( , SqlConnection , SqlTransaction , Int32 )
   en .( )
   en .( )
   en .Execute(List`1 actions)
   en .(List`1 )
   en Z.BulkOperations.BulkOperation.Execute()
   en Z.EntityFramework.Extensions.EntityBulkOperation`1.BulkMerge()
   en .[](BulkOperation`1 this, DbContext , List`1 , Boolean , Type , String )
   en .`1.(SchemaEntityType )
   en System.Collections.Generic.List`1.ForEach(Action`1 action)
   en .[](DbContext this, BulkOperation`1 , IEnumerable`1 )
   en .[](BulkOperation`1 this, DbContext , IEnumerable`1 )
   en DbContextExtensions.BulkMerge[T](DbContext this, IEnumerable`1 entities, Action`1 bulkOperationFactory)
   en DbContextExtensions.BulkMerge[T](DbContext this, IEnumerable`1 entities)
   en IdentitySample.Controllers.HomeController.SyncAD()

The line of code used is:

`db.BulkMerge(lista, options =>  options.ColumnPrimaryKeyExpression = u => u.UserName);`

where lista is  `List<ApplicationUser>`.

It is important for us to know if we can make this work properly since we need to justify its buying with this example.

Regards

**Comment by JonathanMagnan** [comment]
Hello @Borja204 ,

Thank you for reporting.

Could you let me know which version of Entity Framework you are using?

Once we get this information, we will look at it to try to replicate this issue.

Best Regards,

Jonathan

**Comment by Borja204** [comment]
Hello @JonathanMagnan,

I was using Entity Framework 6.1.3. Just updated it to 6.2 and still getting the error.

Regards!

**Comment by Borja204** [comment]
Hello @JonathanMagnan,

You were right, I had this in the ApplicationUser:

`public bool? EmailConfirmed { get; set; }
        public bool? PhoneNumberConfirmed { get; set; }
        public bool? TwoFactorEnabled { get; set; }
        public bool? LockoutEnabled { get; set; }
        public bool? AccessFailedCount { get; set; }`

I know is not the best idea but I just added it to make some properties of the identityUser nullable in a fast way. I've deleted em and make that with the fluent api.

Now works like a charm, bulkMerge worked perfectly and pretty fast for 12k records.

Thanks and regards,
